rs/registry/canister/src/mutations/routing_table.rs

```diff
@@ -279,4 +383,272 @@ mod tests {
 
         // GetSubnetForCanisterError::CanisterIdConversion currently not reachable - CanisterId::try_from() always succeeds
     }
+
+    #[test]
+    fn test_routing_table_saves_as_canister_range_records_on_first_invocation_correctly() {
```

Comment:
I saw this test also in rs/registry/canister/src/registry_lifecycle.rs

Reply:
They're testing slightly different things.  One is testing the migration, which relies on some of the same code, and this tests what happens the first time this function is called.